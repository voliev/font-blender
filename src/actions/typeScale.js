import {
  SET_DEFAULT_TYPE_SCALE_STATE,
  SET_FONT_SIZE,
  SET_TYPE_SCALE
} from "../constants";

export const setTypeScale = (scale = 1.414) => ({
  type: SET_TYPE_SCALE,
  scale
});

export const setFontSize = (fontSize = 16) => ({
  type: SET_FONT_SIZE,
  fontSize
});

export const setDefaultTypeScaleState = () => ({
  type: SET_DEFAULT_TYPE_SCALE_STATE
});
