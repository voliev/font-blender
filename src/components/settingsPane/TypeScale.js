import React, { Component } from "react";
import PropTypes from "prop-types";
import BaseFontSizeInput from "./BaseFontSizeInput";
import ScaleValueSelectList from "./ScaleValueSelectList";
import { numberInputChangeHandler } from "../../utils";
import "./styles/TypeScale.css";

class TypeScale extends Component {
  constructor() {
    super();

    this.handleFontSizeChange = this.handleFontSizeChange.bind(this);
    this.handleScaleChange = this.handleScaleChange.bind(this);
  }

  handleFontSizeChange(e) {
    numberInputChangeHandler(e.target.value, 16, 1, 100, fontSize => {
      this.props.setFontSize(fontSize);
    });
  }

  handleScaleChange(e) {
    const { value } = e.target;
    this.props.setTypeScale(parseFloat(value));
  }

  render() {
    const { disabled, typeScale } = this.props;

    return (
      <div className="TypeScale">
        <p className="group-label">Type Scale</p>
        <BaseFontSizeInput
          disabled={disabled}
          fontSizeChangeHandler={this.handleFontSizeChange}
          value={typeScale.fontSize}
        />
        <ScaleValueSelectList
          disabled={disabled}
          scaleChangeHandler={this.handleScaleChange}
          value={typeScale.scale}
        />
      </div>
    );
  }
}

TypeScale.propTypes = {
  typeScale: PropTypes.exact({
    fontSize: PropTypes.number.isRequired,
    scale: PropTypes.number.isRequired
  }).isRequired,
  disabled: PropTypes.bool.isRequired,
  setFontSize: PropTypes.func.isRequired,
  setTypeScale: PropTypes.func.isRequired
};

export default TypeScale;
