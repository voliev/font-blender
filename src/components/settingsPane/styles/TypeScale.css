.TypeScale {
  margin: 0 var(--space-md);
  padding: var(--space-md) 0;
}

.TypeScale .group-label {
  font-weight: 700;
  margin: 0 0 var(--space-xs) 0;
  text-transform: uppercase;
}
