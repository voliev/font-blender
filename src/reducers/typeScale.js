import { SET_DEFAULT_STATE, SET_FONT_SIZE, SET_TYPE_SCALE } from "../constants";

export const initialState = {
  fontSize: 16,
  scale: 1.414
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_TYPE_SCALE:
      return {
        ...state,
        scale: action.scale
      };
    case SET_FONT_SIZE:
      return {
        ...state,
        fontSize: action.fontSize
      };
    case SET_DEFAULT_STATE:
      return initialState;
    default:
      return state;
  }
};

export const getTypeScale = state => state.styles.present.typeScale;
export const getTypeScaleDefaultState = () => ({ ...initialState });
