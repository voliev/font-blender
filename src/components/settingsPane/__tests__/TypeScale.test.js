import React from "react";
import { mount, shallow } from "enzyme";
import TypeScale from "../TypeScale";
import typeScale from "../../../fixtures/typeScale";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const props = {
    typeScale,
    disabled: false,
    setFontSize: jest.fn(),
    setTypeScale: jest.fn(),
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<TypeScale {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { disabled, setFontSize, setTypeScale, typeScale, wrapper } = setup(
    mount
  );

  expect(typeof wrapper.prop("typeScale")).toBe("object");
  expect(wrapper.prop("typeScale")).toEqual(typeScale);

  expect(typeof wrapper.prop("disabled")).toBe("boolean");
  expect(wrapper.prop("disabled")).toBe(disabled);

  expect(typeof wrapper.prop("setFontSize")).toBe("function");
  expect(wrapper.prop("setFontSize")).toBe(setFontSize);

  expect(typeof wrapper.prop("setTypeScale")).toBe("function");
  expect(wrapper.prop("setTypeScale")).toBe(setTypeScale);
});

it("should handle base font size change", () => {
  const { wrapper, setFontSize } = setup(mount);
  const fontSize = "18";
  const event = {
    target: {
      value: fontSize
    }
  };

  wrapper.find("input").prop("onChange")(event);

  expect(setFontSize).toHaveBeenCalledTimes(1);
  expect(setFontSize).toHaveBeenCalledWith(18);
});

it("should handle scale value change", () => {
  const { wrapper, setTypeScale } = setup(mount);
  const scale = 1.2;
  const event = {
    target: {
      value: scale
    }
  };

  wrapper.find("select").prop("onChange")(event);

  expect(setTypeScale).toHaveBeenCalledTimes(1);
  expect(setTypeScale).toHaveBeenCalledWith(scale);
});
