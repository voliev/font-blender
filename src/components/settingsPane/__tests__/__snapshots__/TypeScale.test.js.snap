// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="TypeScale"
>
  <p
    className="group-label"
  >
    Type Scale
  </p>
  <Memo(BaseFontSizeInput)
    disabled={false}
    fontSizeChangeHandler={[Function]}
    value={16}
  />
  <Memo(ScaleValueSelectList)
    disabled={false}
    scaleChangeHandler={[Function]}
    value={1.414}
  />
</div>
`;
